crates/packaging/src/https.rs

```diff
@@ -102,6 +120,92 @@ impl<'a> PackageMetadata<'a> {
     }
 }
 
+#[test]
+#[should_panic(expected = "MissingHttps")]
+fn url_problem_missing_https() {
+    PackageMetadata::try_from("http://example.com").unwrap();
+}
+
+#[test]
+#[should_panic(expected = "MisleadingCharacter")]
+fn url_problem_misleading_character_at() {
+    PackageMetadata::try_from("https://user:password@example.com/").unwrap();
+}
+
+#[test]
+#[should_panic(expected = "MisleadingCharacter")]
+fn url_problem_misleading_character_unicode_2044() {
+    PackageMetadata::try_from("https://example.com⁄path").unwrap();
+}
+
+#[test]
+#[should_panic(expected = "MisleadingCharacter")]
+fn url_problem_misleading_character_unicode_2215() {
+    PackageMetadata::try_from("https://example.com∕path").unwrap();
+}
+
+#[test]
+#[should_panic(expected = "MisleadingCharacter")]
+fn url_problem_misleading_character_unicode_ff0f() {
+    PackageMetadata::try_from("https://example.com／path").unwrap();
+}
+
+#[test]
+#[should_panic(expected = "MisleadingCharacter")]
+fn url_problem_misleading_character_unicode_29f8() {
+    PackageMetadata::try_from("https://example.com⧸path").unwrap();
```

Comment:
For these, I think it would be clearer to use the actual unicode (e.g. with `\u`) in the string literal, and then show the actual version in a comment above, so you can see why it's misleading.

Reply:
I implemented this suggestion with [778a0dc](https://github.com/roc-lang/roc/pull/5493/commits/778a0dcb67441e29f958b36b0295dd461a6a4b6a), let me know if this is what you had in mind.